app/experimenter/nimbus-ui/src/components/PageEditBranches/FormBranches/index.test.tsx

```diff
@@ -596,6 +599,92 @@ describe("FormBranches", () => {
       expect(screen.getAllByText(FEATURE_VALUE_WARNING)).toHaveLength(1);
     });
   });
+
+  describe("preventPrefConflicts checkbox", () => {
+    it("doesn't render if no feature config set", () => {
+      render(
+        <SubjectBranches
+          experiment={{ ...MOCK_EXPERIMENT, featureConfigs: [] }}
+        />,
+      );
+
+      expect(
+        screen.queryByTestId("prevent-pref-conflicts-checkbox"),
+      ).not.toBeInTheDocument();
+    });
+
+    it("doesn't render if the set feature config does not set prefs", () => {
+      render(
+        <SubjectBranches
+          experiment={{
+            ...MOCK_EXPERIMENT,
+            featureConfigs: [MOCK_FEATURE_CONFIG],
+          }}
+        />,
+      );
+
+      expect(
+        screen.queryByTestId("prevent-pref-conflicts-checkbox"),
+      ).not.toBeInTheDocument();
+    });
+
+    it("does render if the set feature config does not set prefs", () => {
+      render(
+        <SubjectBranches
+          experiment={{
+            ...MOCK_EXPERIMENT,
+            featureConfigs: [MOCK_FEATURE_CONFIG_WITH_SETS_PREFS],
+          }}
+        />,
+      );
+
+      const checkbox = screen.getByTestId(
+        "prevent-pref-conflicts-checkbox",
+      ) as HTMLInputElement;
+      expect(checkbox).toBeInTheDocument();
+      expect(checkbox.checked).toEqual(false);
+    });
+
+    it("resets when switching away from a feature config that sets prefs", () => {
+      render(
+        <SubjectBranches
+          experiment={{
+            ...MOCK_EXPERIMENT,
+            featureConfigs: [MOCK_FEATURE_CONFIG_WITH_SETS_PREFS],
+          }}
+        />,
+      );
+
+      {
+        const checkbox = screen.getByTestId(
+          "prevent-pref-conflicts-checkbox",
+        ) as HTMLInputElement;
+
+        expect(checkbox).toBeInTheDocument();
+        expect(checkbox.checked).toEqual(false);
+
+        fireEvent.click(checkbox);
+        expect(checkbox.checked).toEqual(true);
+      }
+
+      selectFeatureConfig(MOCK_FEATURE_CONFIG.id);
+
+      expect(
+        screen.queryByTestId("prevent-pref-conflicts-checkbox"),
+      ).not.toBeInTheDocument();
+
+      selectFeatureConfig(MOCK_FEATURE_CONFIG_WITH_SETS_PREFS.id);
+
+      {
+        const checkbox = screen.getByTestId(
+          "prevent-pref-conflicts-checkbox",
+        ) as HTMLInputElement;
+
+        expect(checkbox).toBeInTheDocument();
+        expect(checkbox.checked).toEqual(false);
+      }
+    });
```

Comment:
Yeah like I'd treat these two transitions as two separate tests.  Moving from setsprefs to not setsprefs hides the check box, moving from not setsprefs to setsprefs reveals the checkbox.

Reply:
Its testing that if you switch away to not-sets-prefs, the checkbox will be unchecked the next time you select a sets-prefs-feature, which is not the same as a test that switches from not-sets-prefs to sets-prefs (as there is no state to reset)

Reply:
Ah yep that's a good point 👍